petpals/forum/post/routes.py

```diff
@@ -38,3 +38,65 @@ def post_form():
 
     elif request.method == 'GET':
         return render_template('post_form.html', form=form)
+
+
+@router.route('/edit/<int:post_id>', methods=['GET', 'POST'])
+@login_required  # User must be logged in to edit a post
+def edit_post(post_id):
+
+    # Locate Correct Post
+    post = Post.query.get_or_404(post_id)
+    form = NewPostForm()
+    if form.validate_on_submit():
+        post.title = form.title.data
+        post.content = form.content.data
+
+        # Update Database
+        db.session.add(post)
+        db.session.commit()
+
+        # Success Message
+        flash('Post Has Been Updated!', 'success')
+        return redirect(url_for('forum_router.forum', post_id=post.post_id))
+
+    if current_user.id == post.user_id or current_user.id:
+        form.title.data = post.title
+        form.content.data = post.content
+        return render_template('edit_post.html', form=form)
+    else:
+        flash("You Aren't Authorized To Edit This Post...")
+        posts = Post.query.order_by(Post.timestamp)
+        return redirect("forum.html", posts=posts)
+
+
+@router.route('/delete/<int:post_id>')
+@login_required
+def delete_post(post_id):
+    post_to_delete = Post.query.get_or_404(post_id)
+    id = current_user.id
+    if id == post_to_delete.user_id:
+        try:
+            db.session.delete(post_to_delete)
+            db.session.commit()
+
+            # Return a message
+            flash("Blog Post Was Deleted!", 'success')
+
+            # Grab all the posts from the database
+            posts = Post.query.order_by(Post.timestamp)
+            return redirect(url_for('forum_router.forum', posts=posts))
+
+        except:
+            # Return an error message
+            flash("Whoops! There was a problem deleting post, try again...")
+
+            # Grab all the posts from the database
+            posts = Post.query.order_by(Post.timestamp)
+            return redirect('post.html', post_id=post.post_id)
```

Comment:
Invalid redirect syntax, look at other redirect syntax.
With valid syntax, you don't need to query posts.